src/greatwall/resources/greatwall.py

```diff
@@ -234,3 +268,99 @@ def return_level(self):
             self.is_finished = False
         self.current_level -= 1
         self.state = self.protocol_states[self.current_level]
+
+        self.history_step_back()
+        self.history_state_load()
+
+    def history_reset(self):
+
+        self.saved_fractals = {}
+        self.saved_path = []
+        self.saved_states = {}
```

Comment:
It seems that this function is not called anywhere.

Reply:
It was used in a test, but the test was removed. This could have uses because the arity and depth of the tree can change, and when that happens a reset will be required.

Reply:
Also, this is forbidden to introduce an object method that are used only in the test, except some edge case that is not exist here.